lib/taste_tester/client.rb

```diff
@@ -125,14 +125,62 @@ def upload
 
     private
 
+    def tar_content(list, name)
+      full_paths = list.map do |path|
+        "#{TasteTester::Config.repo}/#{path}"
+      end
+      out = full_paths.map do |path|
+        if path.start_with?('/')
+          path = path[1..-1]
+        end
+        "--transform 's,#{path},#{name},'"
+      end
+      out.concat(
+        full_paths.each do |path|
+          "'#{path}'"
+        end,
+      )
+      # partial tar command line + list of full path used for finding
+      # symlinks
+      [out.join(' '), full_paths]
+    end
+
     def full
+      if TasteTester::Config.bundle
+        dest = "#{@server.bundle_dir}/tt.tgz"
+        excludes = []
+        cmd = [
+          [TasteTester::Config.relative_cookbook_dirs, 'cookbooks'],
+          [[TasteTester::Config.relative_role_dir], 'roles'],
+          [[TasteTester::Config.relative_databag_dir], 'databags'],
+        ].map do |list, name|
+          subcmd, paths = tar_content(list, name)
+          excludes.push(paths)
+          subcmd
+        end.join(' ')
+
+        # use find to locate all symlinks so they can be excluded, and pass
+        # into --exclude-from=- (stdin). Maxdepth of 1 so symlinks used within
+        # cookbooks etc are not excluded.
+        cmd = "find #{excludes.join(' ')} -maxdepth 1 -type l | " +
```

Comment:
using cmd to make cmd is ugly. It works because ruby is smart, but use different variables.

Reply:
This was superseded.